### DISCUSSION TITLE
[2405.5.4] Upset Uakari

### DISCUSSION THREAD

**Discussion by nzbr** [nix-community/NixOS-WSL#511] [Announcements]
# Patch Notes

Besides NixOS 24.05, this release brings support for using Nvidia GPUs through the WSL2 integration of their Windows driver.

It will also be the first release to come with Launcher binaries attached. If you want to give that a try, download `NixOS-WSL-Launcher.zip` and extract it. Running `.\NixOS.exe --help` will list the things it is capable of so far. Please report any bugs you encounter. The launcher will be used for distributing NixOS-WSL through additional channels like [scoop](https://scoop.sh) and eventually the Microsoft Store in the future.

syschdemd (the previous method for running systemd in WSL before Microsoft supported that directly) has been deprecated and is scheduled to be removed completely with the 24.11 release. If you have explicitly disabled native systemd in your config, please consider enabling it now.

### Channels

1. Switch your nixpkgs channel to 24.05
    - `sudo nix-channel --add https://nixos.org/channels/nixos-24.05 nixos`
2. Update your channels
    - `sudo nix-channel --update`
3. Rebuild your system
    - `sudo nixos-rebuild switch`
   
### Flakes

If you're using flakes instead of channels, update your `nixpkgs` input to `24.05`, run `nix flake update` and rebuild your system.

# Changelog

## 🛠️ General Changes

* Update nixpkgs to nixos-24.05 (#479) @renovate
* Deprectate syschdemd (#473) @nzbr

## 🚀 Features

* Create /run/booted-system symlink (#325) @SuperSandro2000
* Add shell-wrapper to inject env variables into non-login shells (#452) @nzbr
* Add option to disable error about ldconfig (#443) @SuperSandro2000
* usbip: make more reliable by using default route (#444) @SuperSandro2000
* modules/build-tarball: add e2fsprogs dependency (for chattr), use pigz (#385) @K900
* feat: add option to disable wslg x11 mount (#371) @PhilippHeuer
* wsl-distro: introduce `wsl.useWindowsDriver` option (#339) @Atry

## 🪲 Bugfixes

* fix: make wrapper not crash if logger setup fails (#496) @nzbr
* fix(shell-wrapper): login shell and user systemd (#464) @nzbr
* utils/shell-wrapper: hardcode paths to the right sh/env executables (#459) @K900
* fix: Don't mask resolved systemd unit  (#455) @SuperSandro2000
* fix: prevent systemd from hiding wayland socket (#450) @nzbr
* fix: /tmp/.X11-unix (#432) @nzbr
* fix: bind mount X11 socket instead of symlinking (#420) @nzbr
* fix: set state version when building tarballs (#419) @nzbr
* fix: Create directory for symlink tree (#392) @SuperSandro2000
* fix: Hard-code file names when making symbolic links for drivers (#383) @Atry
* fix: start menu launchers rsync of directory symlinks (#380) @terlar
* fix: symlink creation of per-user apps/icons folder (#368) @mprasil

## 🧪 Tests

* test: Make tests work on Windows with native systemd (#430 #466) @JojOatXGME @nzbr

## 📖 Documentation

* More documentation (#406) @nzbr
* Add instructions to enable WSL 2 (#458) @andoriyu
* Add example flake configuration (#446) @Prince213
* Update README.md - use %userprofile% instead of the cwd (#438) @noamraph

<hr /><em>This discussion was created from the release <a href='https://github.com/nix-community/NixOS-WSL/releases/tag/2405.5.4'>[2405.5.4] Upset Uakari</a>.</em>

**Comment by Nicknamely** [comment]
**NixOS-WSL-Launcher.msixbundle** requires a certificate just like [ArchWSL](https://github.com/yuk7/ArchWSL/releases/tag/24.4.28.0)
